From c815d7cb03d7038f59a6e703c0290b7764a20568 Mon Sep 17 00:00:00 2001
From: Zhang Zekun <zhangzekun11@huawei.com>
Date: Mon, 14 Nov 2022 03:29:46 +0000
Subject: [PATCH 0060/1398] [Huawei] perf: hisi: Add configs for PMU isolation

Offering: HULK
hulk inclusion
category: feature
bugzilla: 187974

------------------------------------------

Add CONFIG_HISI_L3T_PMU and CONFIG_HISI_LPDDRC_PMU to isolate features
of hisi pmu driver.

This patch isolates commit 0edc58409e30 and 6bf896bea639.

Signed-off-by: Zhang Zekun <zhangzekun11@huawei.com>
---
 arch/arm64/configs/openeuler_defconfig |  2 ++
 drivers/perf/hisilicon/Kconfig         | 19 +++++++++++++++++++
 drivers/perf/hisilicon/Makefile        |  7 ++++---
 3 files changed, 25 insertions(+), 3 deletions(-)

diff --git a/arch/arm64/configs/openeuler_defconfig b/arch/arm64/configs/openeuler_defconfig
index d3f6e20f780d..4e3e95fbe3ba 100644
--- a/arch/arm64/configs/openeuler_defconfig
+++ b/arch/arm64/configs/openeuler_defconfig
@@ -6090,6 +6090,8 @@ CONFIG_THUNDERX2_PMU=m
 CONFIG_XGENE_PMU=y
 CONFIG_ARM_SPE_PMU=y
 CONFIG_HISI_PMU=m
+CONFIG_HISI_L3T_PMU=m
+CONFIG_HISI_LPDDRC_PMU=m
 CONFIG_HISI_PCIE_PMU=m
 # end of Performance monitor support
 
diff --git a/drivers/perf/hisilicon/Kconfig b/drivers/perf/hisilicon/Kconfig
index c5d1b7019fff..fbe068bf0100 100644
--- a/drivers/perf/hisilicon/Kconfig
+++ b/drivers/perf/hisilicon/Kconfig
@@ -5,3 +5,22 @@ config HISI_PMU
 	  help
 	  Support for HiSilicon SoC L3 Cache performance monitor, Hydra Home
 	  Agent performance monitor and DDR Controller performance monitor.
+
+config HISI_L3T_PMU
+	tristate "HiSilicon SoC L3T PMU drivers"
+	depends on HISI_PMU
+	default n
+	  help
+	  Support for HiSilicon SoC L3 Cache performance monitor, Hydra Home
+	  Agent performance monitor and DDR Controller performance monitor.
+	  L3T is a specialized PMU driver.
+
+
+config HISI_LPDDRC_PMU
+	tristate "HiSilicon SoC LDPPRC PMU drivers"
+	depends on HISI_PMU
+	default n
+	  help
+	  Support for HiSilicon SoC L3 Cache performance monitor, Hydra Home
+	  Agent performance monitor and DDR Controller performance monitor.
+	  LPDDRC is a specialize PMU driver.
diff --git a/drivers/perf/hisilicon/Makefile b/drivers/perf/hisilicon/Makefile
index 22e384cdfd53..cd4b2c851956 100644
--- a/drivers/perf/hisilicon/Makefile
+++ b/drivers/perf/hisilicon/Makefile
@@ -1,6 +1,7 @@
 # SPDX-License-Identifier: GPL-2.0-only
 obj-$(CONFIG_HISI_PMU) += hisi_uncore_pmu.o hisi_uncore_l3c_pmu.o \
 			  hisi_uncore_hha_pmu.o hisi_uncore_ddrc_pmu.o hisi_uncore_sllc_pmu.o \
-			  hisi_uncore_pa_pmu.o \
-			  hisi_uncore_l3t_pmu.o \
-			  hisi_uncore_lpddrc_pmu.o
+			  hisi_uncore_pa_pmu.o
+
+obj-$(CONFIG_HISI_L3T_PMU) += hisi_uncore_l3t_pmu.o
+obj-$(CONFIG_HISI_LPDDRC_PMU) += hisi_uncore_lpddrc_pmu.o
-- 
2.17.1

